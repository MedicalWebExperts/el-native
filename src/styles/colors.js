export default {
  primary: 'lightgreen',
  blueColor: '#2196F3',
  lightgray: 'lightgray',
  white: 'white',
  red: 'red',
  textColor: '#757575',
  darkTextColor: '#757575',
};
