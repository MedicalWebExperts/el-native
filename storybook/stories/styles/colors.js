export default {
  primary: 'lightgreen',
  backgroundLight: '#F5FCFF',
  backgroundDark: '#AFAFAF',
  customColor: 'aquamarine',
  blueColor: '#2196F3',
  lightgray: 'lightgray',
  darkgray: 'gray',
  white: 'white',
};
