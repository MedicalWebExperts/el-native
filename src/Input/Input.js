import React, { Component } from 'react';
import { string, number, bool, oneOf, object } from 'prop-types';
import { Input as RNInput } from 'react-native';

import colors from '../styles/colors';

class Input extends Component {
  static propTypes = {
    /**
     * What Input to show, see Input Explorer app or one of the links above. Default: heart.
     */
    name: string,
    /**
     * Color of the Input. Default: Theme white color.
     */
    value: string,
    /**
     * Shows a placeholder with the provided value.
     */
    placeholder: string,
    /**
     * This prop expects a function that will be executed when the user types.
     */
    onChangeText: func,
    /**
     * Shows the Input as a Text Area.
     */
    textArea: bool,
    /**
     * Determines the number of lines of the Text Area. The default is 4.
     */
    numberOfLines: number,
    /**
     * Boolean that allows the Input to autocorrect the text user is typing, defaulted to "false".
     */
    autoCorrect: bool,
    /**
     * Determines whether the Input is readonly. The default is "false".
     */
    editable: bool,
    /**
     * Selects text on focus with this prop. Default is "false".
     */
    selectTextOnFocus: bool,
    /**
     * Shows predefined error styling when passed "true". Defaulted to "false".
     */
    error: bool,
    /**
     * Shows a provided text under the Input to give more clarity on the kind of error.
     * If prop not used, no text will appear.
     */
    errorText: string,
    /**
     * Prop for custom styling.
     */
    style: object,
    /**
     * Predefined shapes, our customized Inputs. It could be one of "underline", "square" or
     * "rounded". The default is "underline".
     */
    type: oneOf(['underline', 'square', 'rounded']),
  };

  static defaultProps = {
    name: 'ios-heart',
    value: '',
    placeholder: 'Placeholder text',
    onChangeText: () => {},
    textArea: false,
    numberOfLines: 4,
    autoCorrect: false,
    editable: false,
    selectTextOnFocus: false,
    error: false,
    errorText: 'Error Text',
    style: {},
    type: 'underline',
  };

  /**
   * Input component is a wrapper over React Native Input with 3 predefined styles added
   * and the possibility of further customizing.
   */

  handleChangeContent = (text) => {};

  render() {
    const { value } = this.state;
    const {
      name,
      placeholder,
      textArea,
      numberOfLines,
      autoCorrect,
      editable,
      selectTextOnFocus,
      error,
      errorText,
      style,
      type,
    } = this.props;

    return (
      <RNInput
        name={name}
        value={value}
        placeholder={placeholder}
        onChangeText={text => this.handleChangeContent(text)}
        multiline={textArea}
        numberOfLines={numberOfLines}
        autoCorrect={autoCorrect}
        editable={editable}
        selectTextOnFocus={selectTextOnFocus}
        error={error}
        errorText={errorText}
        style={style}
        type={type}
      />
    );
  }
}

Input.propTypes = propTypes;
Input.defaultProps = defaultProps;

export default Input;
