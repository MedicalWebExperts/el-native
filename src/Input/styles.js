import colors from '../../src/styles/colors';

export default {
  base: {
    borderColor: colors.textColor,
  },
  error: {
    borderColor: colors.red,
  },
  underline: {},
  squared: { borderColor: colors.red },
  rounded: { borderColor: colors.red },
};
