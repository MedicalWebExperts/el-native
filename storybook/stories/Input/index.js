import React from 'react';
import { storiesOf } from '@storybook/react-native';

import CenterView from '../CenterView';
import { Input } from '../../../src';
import colors from '../styles/colors';

storiesOf('Input', module)
  .addDecorator(getStory => (
    <CenterView style={{ backgroundColor: colors.backgroundDark }}>{getStory()}</CenterView>
  ))
  .add('Default', () => <Input />)
  .add('Custom color', () => <Input size={200} color={colors.customColor} />)
  .add('Custom size', () => <Input size={150} />)
  .add('Custom Input', () => <Input size={200} name="ios-basketball" />)
  .add('Custom all', () => <Input name="ios-football" size={250} color={colors.customColor} />);
