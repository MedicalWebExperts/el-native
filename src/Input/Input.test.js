import React from 'react';
import renderer from 'react-test-renderer';

import Input from './Input';
import colors from '../styles/colors';

const props = {
  placeholder: 'Placeholder text',
  onChangeText: () => {},
  multiline: false,
  numberOfLines: 4,
  autoCorrect: false,
  editable: false,
  selectTextOnFocus: false,
  error: false,
  errorText: 'Error text',
  style: {},
  type: 'underline',
  placeholderTextColor: colors.textColor,
};

const customStyle = {
  borderColor: colors.primary,
};

describe('Input Snapshots', () => {
  it('should render an Input default', () => {
    const component = renderer.create(<Input {...props} />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
  it('should render an Input default with error', () => {
    const component = renderer.create(<Input {...props} error />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
  it('should render an Input with custom styling', () => {
    const component = renderer.create(<Input {...props} style={customStyle} />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
  it('should render a squared Input', () => {
    const component = renderer.create(<Input {...props} style={customStyle} type="squared" />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
  it('should render a squared Input with error', () => {
    const component = renderer.create(
      <Input {...props} style={customStyle} type="squared" error />,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
  it('should render a rounded Input', () => {
    const component = renderer.create(<Input {...props} style={customStyle} type="rounded" />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
  it('should render a rounded Input with error', () => {
    const component = renderer.create(
      <Input {...props} style={customStyle} type="rounded" error />,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
  it('should render a Text Area Input', () => {
    const component = renderer.create(
      <Input {...props} style={customStyle} placeholder="Text Area" multiline numberOfLines={8} />,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
