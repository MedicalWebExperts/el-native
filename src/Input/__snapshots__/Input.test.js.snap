// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Snapshots should render a Text Area Input 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "flexDirection": "column",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    autoCorrect={false}
    editable={false}
    multiline={true}
    numberOfLines={8}
    onChangeText={[Function]}
    placeholder="Text Area"
    placeholderTextColor="#757575"
    selectTextOnFocus={false}
    style={
      Array [
        Object {
          "borderBottomWidth": 1,
          "borderColor": "#757575",
          "color": undefined,
          "marginTop": 4,
          "minHeight": 40,
          "textAlign": "left",
          "width": 750,
        },
        Object {
          "borderColor": "lightgreen",
        },
      ]
    }
    value=""
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "red",
        "fontSize": 12,
        "paddingVertical": 4,
      }
    }
  />
</View>
`;

exports[`Input Snapshots should render a rounded Input 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "flexDirection": "column",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    autoCorrect={false}
    editable={false}
    multiline={false}
    numberOfLines={4}
    onChangeText={[Function]}
    placeholder="Placeholder text"
    placeholderTextColor="#757575"
    selectTextOnFocus={false}
    style={
      Array [
        Object {
          "borderBottomWidth": 1,
          "borderColor": "#757575",
          "borderRadius": 20,
          "borderWidth": 1,
          "color": undefined,
          "marginTop": 4,
          "minHeight": 40,
          "paddingLeft": 10,
          "textAlign": "left",
          "width": 750,
        },
        Object {
          "borderColor": "lightgreen",
        },
      ]
    }
    value=""
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "red",
        "fontSize": 12,
        "paddingVertical": 4,
      }
    }
  />
</View>
`;

exports[`Input Snapshots should render a rounded Input with error 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "flexDirection": "column",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    autoCorrect={false}
    editable={false}
    multiline={false}
    numberOfLines={4}
    onChangeText={[Function]}
    placeholder="Placeholder text"
    placeholderTextColor="#757575"
    selectTextOnFocus={false}
    style={
      Array [
        Object {
          "borderBottomWidth": 1,
          "borderColor": "red",
          "borderRadius": 20,
          "borderWidth": 1,
          "color": undefined,
          "marginTop": 4,
          "minHeight": 40,
          "paddingLeft": 10,
          "textAlign": "left",
          "width": 750,
        },
        Object {
          "borderColor": "lightgreen",
        },
      ]
    }
    value=""
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "red",
        "fontSize": 12,
        "paddingVertical": 4,
      }
    }
  >
    Error text
  </Text>
</View>
`;

exports[`Input Snapshots should render a squared Input 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "flexDirection": "column",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    autoCorrect={false}
    editable={false}
    multiline={false}
    numberOfLines={4}
    onChangeText={[Function]}
    placeholder="Placeholder text"
    placeholderTextColor="#757575"
    selectTextOnFocus={false}
    style={
      Array [
        Object {
          "borderBottomWidth": 1,
          "borderColor": "#757575",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": undefined,
          "marginTop": 4,
          "minHeight": 40,
          "paddingLeft": 10,
          "textAlign": "left",
          "width": 750,
        },
        Object {
          "borderColor": "lightgreen",
        },
      ]
    }
    value=""
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "red",
        "fontSize": 12,
        "paddingVertical": 4,
      }
    }
  />
</View>
`;

exports[`Input Snapshots should render a squared Input with error 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "flexDirection": "column",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    autoCorrect={false}
    editable={false}
    multiline={false}
    numberOfLines={4}
    onChangeText={[Function]}
    placeholder="Placeholder text"
    placeholderTextColor="#757575"
    selectTextOnFocus={false}
    style={
      Array [
        Object {
          "borderBottomWidth": 1,
          "borderColor": "red",
          "borderRadius": 5,
          "borderWidth": 1,
          "color": undefined,
          "marginTop": 4,
          "minHeight": 40,
          "paddingLeft": 10,
          "textAlign": "left",
          "width": 750,
        },
        Object {
          "borderColor": "lightgreen",
        },
      ]
    }
    value=""
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "red",
        "fontSize": 12,
        "paddingVertical": 4,
      }
    }
  >
    Error text
  </Text>
</View>
`;

exports[`Input Snapshots should render an Input default 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "flexDirection": "column",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    autoCorrect={false}
    editable={false}
    multiline={false}
    numberOfLines={4}
    onChangeText={[Function]}
    placeholder="Placeholder text"
    placeholderTextColor="#757575"
    selectTextOnFocus={false}
    style={
      Array [
        Object {
          "borderBottomWidth": 1,
          "borderColor": "#757575",
          "color": undefined,
          "marginTop": 4,
          "minHeight": 40,
          "textAlign": "left",
          "width": 750,
        },
        Object {},
      ]
    }
    value=""
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "red",
        "fontSize": 12,
        "paddingVertical": 4,
      }
    }
  />
</View>
`;

exports[`Input Snapshots should render an Input default with error 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "flexDirection": "column",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    autoCorrect={false}
    editable={false}
    multiline={false}
    numberOfLines={4}
    onChangeText={[Function]}
    placeholder="Placeholder text"
    placeholderTextColor="#757575"
    selectTextOnFocus={false}
    style={
      Array [
        Object {
          "borderBottomWidth": 1,
          "borderColor": "red",
          "color": undefined,
          "marginTop": 4,
          "minHeight": 40,
          "textAlign": "left",
          "width": 750,
        },
        Object {},
      ]
    }
    value=""
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "red",
        "fontSize": 12,
        "paddingVertical": 4,
      }
    }
  >
    Error text
  </Text>
</View>
`;

exports[`Input Snapshots should render an Input with custom styling 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-start",
      "flexDirection": "column",
    }
  }
>
  <TextInput
    allowFontScaling={true}
    autoCorrect={false}
    editable={false}
    multiline={false}
    numberOfLines={4}
    onChangeText={[Function]}
    placeholder="Placeholder text"
    placeholderTextColor="#757575"
    selectTextOnFocus={false}
    style={
      Array [
        Object {
          "borderBottomWidth": 1,
          "borderColor": "#757575",
          "color": undefined,
          "marginTop": 4,
          "minHeight": 40,
          "textAlign": "left",
          "width": 750,
        },
        Object {
          "borderColor": "lightgreen",
        },
      ]
    }
    value=""
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "red",
        "fontSize": 12,
        "paddingVertical": 4,
      }
    }
  />
</View>
`;
